packages/hasura-auth-js/tests/utils.test.ts

```diff
@@ -1,82 +1,266 @@
+    expect(isValidEmail([])).toBe(false)
+    expect(isValidEmail({})).toBe(false)
+  })
 })
 
-test(`should keep the trailing / in the clientUrl`, async () => {
-  const options: RedirectOption = { redirectTo: '/' }
+suite('isValidPassword', () => {
+  test('should return true for valid passwords', () => {
+    expect(isValidPassword('abc123')).toBe(true)
+    expect(isValidPassword('abcdefghijklmnopqrstuvwxyz')).toBe(true)
+    expect(isValidPassword('ABCDEFGHIJKLMNOPQRSTUVWXYZ')).toBe(true)
+    expect(isValidPassword('1234567890')).toBe(true)
+    expect(isValidPassword('!@#$%^&*')).toBe(true)
+    expect(isValidPassword('abc123ABC!@#$')).toBe(true)
+  })
 
-  const result = rewriteRedirectTo('https://frontend.com/', options)
-  expect(result?.redirectTo).toEqual('https://frontend.com/')
+  test('should return false for invalid passwords', () => {
+    expect(isValidPassword('')).toBe(false)
+    expect(isValidPassword('ab')).toBe(false)
+    expect(isValidPassword(null)).toBe(false)
+    expect(isValidPassword(undefined)).toBe(false)
+    expect(isValidPassword(12)).toBe(false)
+    expect(isValidPassword(123123)).toBe(false)
+    expect(isValidPassword([])).toBe(false)
+    expect(isValidPassword({})).toBe(false)
+  })
 })
 
-test(`should extend to a relative URL when the clientUrl ends with a slash`, async () => {
-  const options: RedirectOption = { redirectTo: '/nested' }
+suite('getParameterByName', () => {
+  test('should return the value of the specified parameter', () => {
+    expect(getParameterByName('param', 'http://example.com?param=value')).toBe('value')
+    expect(getParameterByName('param', 'http://example.com?other=other&param=value')).toBe('value')
+    expect(getParameterByName('param', 'http://example.com?param=value&other=other')).toBe('value')
+    expect(getParameterByName('param', 'http://example.com?param=value#hash')).toBe('value')
+  })
 
-  const result = rewriteRedirectTo('https://frontend.com/main/', options)
-  expect(result?.redirectTo).toEqual('https://frontend.com/main/nested')
+  test('should return an empty string for a parameter with no value', () => {
+    expect(getParameterByName('param', 'http://example.com?param')).toBe('')
+    expect(getParameterByName('param', 'http://example.com?param=')).toBe('')
+  })
+
+  test('should return null for a missing parameter', () => {
+    expect(getParameterByName('param', 'http://example.com')).toBe(null)
+    expect(getParameterByName('param', 'http://example.com?other=value')).toBe(null)
+  })
+
+  test('should return the value of the specified parameter after the hash', () => {
+    expect(getParameterByName('param', 'http://example.com#param=value')).toBe('value')
+    expect(getParameterByName('param', 'http://example.com#other=other&param=value')).toBe('value')
+    expect(getParameterByName('param', 'http://example.com#param=value&other=other')).toBe('value')
+  })
 })
+
+// suite('removeParameterFromWindow', () => {
```

Comment:
We should either explain why this part is commented, or remove it

Reply:
totally

Reply:
got it working now with jsdom